src/data/rat/defs.lean

```diff
@@ -465,10 +466,13 @@ instance : comm_ring ℚ :=
   mul_assoc        := rat.mul_assoc,
   left_distrib     := rat.mul_add,
   right_distrib    := rat.add_mul,
-  nat_cast         := λ n, rat.of_int n,
+  int_cast         := coe,
+  /- Important: We do not set `nat_cast := λ n, ((n : ℤ) : ℚ)` (even though it's defeq) as that
+  makes `int.cast_coe_nat` and `coe_coe` loop in `simp`. -/
```

Comment:
Sounds to me like `int.cast_coe_nat` is a bad simp lemma.

Reply:
I doubt so, because you need to normalise somehow and you definitely don't want `((n : ℤ) : α)` to unfold to `(((n : ℤ) : ℚ) : α)`. The problem is that we still support transitive coercions and need `coe_coe` to unfold them.